### DISCUSSION TITLE
Disambiguate the 'free' function variable

### DISCUSSION THREAD

**Discussion by FullValueRider** [twinbasic/twinbasic#765] [Ideas]
For VBA functions we get  a predeclared variable of the same name and Type as the Function Name.  
However the usage of this variable can be confusing.

It would be helpful if this variable could be disambiguated and presented as a variable named 'Result' or _Name (where Name is the Method name).

**Comment by bclothier** [comment]
Why not just use `Return` statement instead?

**Comment by FullValueRider** [comment]
Because I'm not returning a value at that stage, but manipulating/populating the 'result'.  The use of the result variable would only save one line of code used to declare a temporary variable so its not that important.

**Reply by bclothier** [reply]
In that case, I'd prefer to declare the variable explicitly and return that. I would assume that the compiler either is or will be smart to optimize out the unused variable and we gain a clearer & readable code in the process.
